src/symbol/quads.js

```diff
@@ -71,7 +71,7 @@ function getIconQuads(anchor: Anchor,
     if (layout['icon-text-fit'] !== 'none' && shapedText) {
         const iconWidth = (right - left),
             iconHeight = (bottom - top),
-            size = layout['text-size'] / 24,
+            size = layer.getLayoutValue('text-size', globalProperties, feature) / 24,
```

Comment:
Why isn't this using `{zoom: bucket.zoom + 1}` like we do in `symbol_layout#addFeature`? That seems like the best match. @anandthakker understands that logic better than I do, though, I think.

Reply:
Woah, good point @ChrisLoer.   This raises an interesting issue.  It's not just that we need to evaluate `text-size` with `{zoom: bucket.zoom + 1}` during layout (though I think you're right about that) -- if text-size is zoom depentent, we also need to scale the icon at render time according to the current text-size value.  And if text-size is a composite function, that's going to be problematic -- we _could_ try to use the text-size attributes for the icon rendering pass in that case, but that would only help for `icon-text-size: both`.

In any case, I think this goes well beyond the scope of this PR.  I'll ticket it separately, 👍 for now!

Reply:
https://github.com/mapbox/mapbox-gl-js/issues/5656